addons/mail/security/mail_security.xml

```diff
@@ -96,6 +96,52 @@
             <field name="perm_unlink" eval="False"/>
         </record>
 
+        <record id="ir_rule_discuss_channel_member_is_channel_owner_unlink" model="ir.rule">
+            <field name="name">discuss.channel.member: channel owner can remove members except owners</field>
+            <field name="model_id" ref="mail.model_discuss_channel_member"/>
+            <field name="groups"
+                eval="[
+                    Command.link(ref('base.group_user')),
+                    Command.link(ref('base.group_portal')),
+                    Command.link(ref('base.group_public')),
+                ]"
+            />
+            <field name="domain_force">
+                [
+                    ("channel_id.self_member_id.channel_role", "=", "owner"),
+                    ("channel_role", "!=", "owner"),
+                    "|",
+                        ("channel_id.channel_type", "=", "channel"),
+                        ("channel_id.channel_type", "=", "group"),
+                ]
+            </field>
+            <field name="perm_create" eval="False"/>
+            <field name="perm_read" eval="False"/>
+            <field name="perm_write" eval="False"/>
+        </record>
+        <record id="ir_rule_discuss_channel_member_is_channel_admin_unlink" model="ir.rule">
+            <field name="name">discuss.channel.member: channel admin can remove members except admins and owners</field>
+            <field name="model_id" ref="mail.model_discuss_channel_member"/>
+            <field name="groups"
+                eval="[
+                    Command.link(ref('base.group_user')),
+                    Command.link(ref('base.group_portal')),
+                    Command.link(ref('base.group_public')),
+                ]"
+            />
+            <field name="domain_force">
+                [
+                    ("channel_id.self_member_id.channel_role", "=", "admin"),
+                    ("channel_role", "in", ["owner", "admin"]),
```

Comment:
Shouldn't the condition be channel_role = false instead?
admin can remove member only if the target is not owner and not admin